src/rdb_protocol/terms/string.cc

```diff
@@ -125,6 +127,71 @@ class match_term_t : public op_term_t {
     virtual const char *name() const { return "match"; }
 };
 
+class format_term_t : public op_term_t {
+public:
+    format_term_t(compile_env_t *env, const raw_term_t &term)
+            : op_term_t{env, term, argspec_t{2}}, compile_env{env} {}
+private:
+    virtual scoped_ptr_t <val_t> eval_impl(scope_env_t *env, args_t *args, eval_flags_t) const {
+        const std::string templ = args->arg(env, 0)->as_str().to_std();
+        const datum_t datum = args->arg(env, 1)->as_datum();
+
+        const size_t template_length = templ.size();
+        std::string formatted{};
+
+        for (size_t pos = 0; pos < template_length; ++pos) {
+            rcheck(templ[pos] != '}',
+                   base_exc_t::LOGIC,
+                   "No parameter tags to close");
+
+            if (templ[pos] == '{') {
+                size_t param_end_pos {1};
+                std::string param_name {};
+
+                while (pos + param_end_pos < template_length && templ[pos + param_end_pos] != '}') {
+                    char next_char{templ[pos + param_end_pos]};
+
+                    rcheck(next_char != '{',
+                           base_exc_t::LOGIC,
+                           "Nested template parameters are not allowed");
+
+                    // Reserve "," for later use to implement formatting
+                    rcheck(next_char != ':' && next_char != '\\',
+                           base_exc_t::LOGIC,
+                           strprintf("Formatting separator \"%c\" is not allowed in parameter name", next_char));
```

Comment:
Actually, one more thing.  The other error messages all use backtick characters for quoting variable names as well as user-supplied strings so it should be consistent with that.  Also, backslash isn't a formatting separator, so, I changed its message.

I looked and some rcheck lines had a trailing period character `.` on the error messages, but others did not.  So I guess that can go either way.

```suggestion
                    // Reserve ":" for later use to implement formatting, "\\" for escaping.
                    rcheck(next_char != ':',
                           base_exc_t::LOGIC,
                           strprintf("Formatting separator `%c` is not allowed in parameter name", next_char));
                    rcheck(next_char != '\\',
                           base_exc_t::LOGIC,
                           strprintf("Reserved escape character `%c` is not allowed in parameter name", next_char));

```